src/Domain/Value/Translation/TranslatableInterface.php

```diff
@@ -0,0 +1,33 @@
+<?php declare(strict_types=1);
+
+namespace PhpTuf\ComposerStager\Domain\Value\Translation;
+
+use PhpTuf\ComposerStager\Domain\Service\Translation\TranslatorInterface;
+use Stringable;
+
+/**
+ * Handles a translatable message.
+ *
+ * This interface is modeled after the Symfony Translation component. However,
+ * there is no guarantee of functional equivalence. Do not depend on undocumented
+ * behavior.
+ *
+ * @see https://symfony.com/doc/current/translation.html
+ *
+ * @package Translation
+ *
+ * @api
+ */
+interface TranslatableInterface extends Stringable
+{
+    /**
+     * Translates the message.
+     *
+     * @throws \PhpTuf\ComposerStager\Domain\Exception\InvalidArgumentException
+     *   If `$locale` contains invalid characters.
+     */
+    public function trans(TranslatorInterface $translator, ?string $locale = null): string;
+
+    /** Returns the bare message as-given, untranslated, without placeholder substitution. */
+    public function __toString(): string;
```

Comment:
This requirement of __toString() is incompatible with Drupal's TranslatableMarkup, because Drupal's TranslatableMarkup::__toString() returns the translated string (it basically does the same as trans()).

Can we change this Composer Stager interface so that instead of defining __toString() we define a different method (e.g., noTrans()) to return the string in its original language? This can be deferred to a followup issue instead of blocking an initial merge of this PR.